docs/cue/reference/components.cue

```diff
@@ -879,6 +943,82 @@ components: {
 			}
 		}
 
+		env_vars: {
+			_http_proxy: {
+				description: """
+					The global URL to proxy HTTPS requests through.
+
+					If another HTTPS proxy is set in the configuration file or at a component level,
+					this one will be overridden.
+
+					The lowercase variant has priority over the uppercase one.
+					"""
+				type: string: {
+					default: null
+					examples: ["http://foo.bar:3128"]
+					syntax: "literal"
+				}
+			}
+			_https_proxy: {
+				description: """
+					The global URL to proxy HTTPS requests through.
+
+					If another HTTPS proxy is set in the configuration file or at a component level,
+					this one will be overriden.
+
+					The lowercase variant has priority over the uppercase one.
+					"""
+				type: string: {
+					default: null
+					examples: ["http://foo.bar:3128"]
+					syntax: "literal"
+				}
+			}
+			_no_proxy: {
+				description: """
+					List of hosts to avoid proxying globally.
+
+					Allowed patterns here include:
+						- Domain names. For example, `example.com` will match requests to to `example.com`
+						- Wildcard domains. For example, `.example.com` will match requests to `example.com` and its subdomains
+						- IP addresses. For example, `127.0.0.1` will match requests to 127.0.0.1
+						- CIDR blocks. For example, `192.168.0.0./16` will match requests to any IP addresses in this range
+						- `*` will match all hosts
+
+					If another no_proxy value is set in the configuration file or at a component level, this one will be overridden.
+
+					The lowercase variant has priority over the uppercase one.
+					"""
+				type: string: {
+					default: null
+					examples: ["localhost,.foo.bar", "*"]
+					syntax: "literal"
+				}
+			}
+			if features.collect != _|_ {
+				if features.collect.proxy != _|_ {
+					if features.collect.proxy.enabled {
+						http_proxy:  env_vars._http_proxy
```

Comment:
I would prefer that we not support the lower case variant. Is this common? None of our other supported env vars accept case variants.

Reply:
It is common specifically for these proxy environment variables to be lower-cased. It is a bit anomalous but dates back to their introduction. See https://github.com/timberio/vector/pull/8205#discussion_r678533122 for more conversation.

Reply:
Could we just not document it and look for lower case variants of all env vars? It seems like noise to me to add both in the list of supported env vars. At the very least, we should have a `variants` field or something.

(this is not blocking)

Reply:
Yeah, or maybe we could just have a note under the uppercase definitions that the lowercase definition is also recognized. I do realize it is a bit of bloat.

Alternatively, maybe we could expand our cue files to allow env var aliases? That could come in handy in other situations too.

(also not blocking)